pkg/vcs/vcs.go

```diff
@@ -58,3 +85,117 @@ func VCS(path string) (result v1.VCS) {
 
 	return
 }
+
+func ImageInfoFromApp(ctx context.Context, app *apiv1.App, cloneDir string) (string, string, error) {
+	vcs := app.Status.AppImage.VCS
+	if len(vcs.Remotes) == 0 {
+		return "", "", fmt.Errorf("clone can only be done on an app built from a git repository")
+	}
+	if vcs.Acornfile == "" || vcs.BuildContext == "" {
+		return "", "", fmt.Errorf("app is missing required vcs information, image must be rebuilt with a newer acorn cli")
+	}
+
+	for _, remote := range vcs.Remotes {
+		// Determine the repository name from the repo url
+		idx := strings.LastIndex(remote, "/")
+		if idx < 0 || idx >= len(remote)-1 {
+			fmt.Printf("failed to determine repository name %q\n", remote)
+			continue
+		}
+		workdir := filepath.Join(cloneDir, strings.TrimSuffix(remote[idx+1:], ".git"))
```

Comment:
Thinking through this a bit more, I think the dir we are cloning into should be the app name, not the repo name. I think that will be most "obvious" to the user. It would also let me o `acorn dev --clone` for two different applications without worrying about collisions

Reply:
When you make this change (and Im happy to debate it if you or @thedadams dont agree with it), some other logic should change. You should check for the existence of the dir outside this for loop and error out if it exists prior to the clone.

Reply:
I think this makes sense and will avoid some of this awkward string processing.

Reply:
Can do